src/Nethermind/Nethermind.HonestValidator/ValidatorClient.cs

```diff
@@ -115,6 +109,54 @@ public Root ComputeSigningRoot(Root objectRoot, Domain domain)
             return _cryptographyService.HashTreeRoot(domainWrappedObject);
         }
 
+        public ulong GetAggregationTime(BeaconChainInformation beaconChainInformation, Slot slot)
+        {
+            var timeParameters = _timeParameterOptions.CurrentValue;
+            ulong startTimeOfSlot = beaconChainInformation.GenesisTime + timeParameters.SecondsPerSlot * slot;
+
+            // Aggregate 2/3 way through slot
+            ulong aggregationTime = startTimeOfSlot + (_timeParameterOptions.CurrentValue.SecondsPerSlot * 2 / 3);
+
+            return aggregationTime;
+        }
+
+        public BlsSignature GetAttestationSignature(Attestation unsignedAttestation, BlsPublicKey blsPublicKey)
+        {
+            var fork = _beaconChainInformation.Fork;
+            var epoch = ComputeEpochAtSlot(unsignedAttestation.Data.Slot);
+
+            ForkVersion forkVersion;
+            if (epoch < fork.Epoch)
+            {
+                forkVersion = fork.PreviousVersion;
+            }
+            else
+            {
+                forkVersion = fork.CurrentVersion;
+            }
+
+            var domainType = _signatureDomainOptions.CurrentValue.BeaconAttester;
+            var attesterDomain = ComputeDomain(domainType, forkVersion);
```

Comment:
if the fork does not change often I think we should definitely cache it (since domain requires the root calculation)